enhancements/cluster-logging/loki-storage.md

```diff
@@ -0,0 +1,628 @@
+
+* As a cluster logging admin, I want to select a different scale for my Loki deployment to enable single redundancy and auto-compaction.
+
+* As a cluster logging admin, I want to expand resources/limits for individual Loki components to support load scenarios not declared by the cluster t-shirt size selected.
+
+* As a cluster logging admin, I want to configure scheduling of individual Loki components to custom selected nodes on my cluster to separate resource usage of the cluster logging from other workloads on my cluster.
+
+* As a Cluster Logging admin, I want to tune the ingestion limits globally or per tenant to support custom ingestion characteristics (e.g. ingenstion rate, label cardinality).
+
+* As a Cluster Logging admin, I want to tune the query limits globally or per tenant to support custom query characteristics (e.g. chunks per series returned).
+
+### Implementation Details
+
+#### Assumptions
+
+* The proposed API will be owned and managed exclusively as a custom resource definition by the [loki-operator](https://github.com/viaq/loki-operator).
+
+* The [cluster-logging-operator](https://github.com/openshift/cluster-logging-operator) will only create a stanza of this CRD and in addition manage an owner reference on it.
+
+#### Security
+
+The cluster logging admin is responsible for creating and maintaining the secret in the `openshift-logging` namespace to permit access for the Loki deployment to connect securely to an object storage endpoint.
+
+The secret is required to provide all of the following key-value pairs:
+- `bucketnames`: Comma separated list of bucket names to evenly distribute chunks over.
+- `endpoint`: S3-compatible API Endpoint to connect to.
+- `access_key_id`: Access Key ID.
+- `access_key_secret`: Secret Access Key.
+
+Optional:
+- `region`: Region to use.
+
+### Risks and Mitigations
+
+In general the key risks to provide the proposed API in form of a custom resource definition are:
+
+**Risk:** Providing too many configuration knobs and in turn a brittle API for managing Loki deployments.
+
+**Mitigation:** Limit by maintaining a strong opinionated API for Loki deployments that encompasses only features around dependencies (e.g. object storage), scheduling (e.g. requests/limits, tolerations/node-selectors) and no more.
+
+**Risk:** Providing poor user experience for day 1 and day 2, e.g. user needs to figure out requests/limits for each component.
+
+**Mitigation:** Provide opinionated and battle-tested scale out formats in form of a T-Shirt size (e.g. `1x.medium`). Leaving room for adding more formats later (e.g. `2x.medium` for hard multi-tenancy) and knobs (e.g. requests/limits) to adapt the format to live environments.
+
+## Design Details
+
+The discussion about the proposed API is split in two sections. The first section `lokistack_types.go` describes the custom resource definition for the [loki-operator](https://github.com/viaq/loki-operator) to reconcile a Loki deployment.
```

Comment:
This is an interesting point, thanks for raising.

The discussion in the log exploration team to use `LokiStack` instead of `Loki` is based on two observations:
- To operate Loki for various scenarios (e.g. authn/authz), we might introduce auxiliary components, like the [observatorium-api](https://github.com/observatorium/observatorium). Thus we cannot speak anymore about Loki, but more of a stack that includes Loki.
- The term `LokiStack` is also a little bit derived from Grafana's [helm charts](https://grafana.com/docs/loki/latest/installation/helm/). They refer to a `LokiStack` every time you do install Loki plus other things, like Grafana/Prometheus/Promtail. We don't intend to install any of those, but for our Gateway proxy case this sound like being well aligned.

WDYT?

Reply:
Stack refers to the components that make up the whole. Each component cannot be useful on its own. It is a delineation between the single-store instance of Loki - i.e. the all-in-one system which is intended for development and testing.

Reply:
Still seems redundant. Loki is _always_ composed of distributed components. Even if you choose to "distribute" them all into a single container in single-store mode, they are still a "stack". So it seems to me that the word "Loki" already means "set of distributed components", and there's no need to add "Stack". But this is a nit, not something to hold up the PR.

Reply:
Also this sounds like a counter argument: "The term LokiStack is also a little bit derived from Grafana's helm charts. They refer to a LokiStack every time you do install Loki plus other things, like Grafana/Prometheus/Promtail." 
We explicitly do not want this operator to manage collectors or visualization, so it sounds like LokiStack is a name with built-in scope-creep.

Reply:
As per discussion above we will stick with `LokiStack` for now to keep the option to add auxiliary components for Loki in the stack.